jest.config.js

```diff
@@ -30,22 +30,30 @@ const customJestConfig = {
 // This won't count for the retry to avoid duplicated test being reported twice
 // - which means our test trace will report test results for the flaky test as failed without retry.
 const shouldEnableTestTrace =
-  process.env.DATADOG_API_KEY &&
-  process.env.DATADOG_TRACE_NEXTJS_TEST &&
+  !!process.env.DATADOG_API_KEY &&
+  !!process.env.DATADOG_TRACE_NEXTJS_TEST &&
   !process.env.IS_RETRY
 
+console.log('-------------', {
+  shouldEnableTestTrace,
+  retry: process.env.IS_RETRY
+})
+
 if (shouldEnableTestTrace) {
   if (!customJestConfig.reporters) {
     customJestConfig.reporters = ['default']
   }
 
-  const outputDirectory = process.env.TURBOPACK
-    ? '<rootDir>/turbopack-test-junit-report'
-    : '<rootDir>/test-junit-report'
+  const outputDirectory =
+    process.env.TURBOPACK || process.env.EXPERIMENTAL_TURBOPACK
+      ? '<rootDir>/turbopack-test-junit-report'
+      : '<rootDir>/test-junit-report'
+
   customJestConfig.reporters.push([
     'jest-junit',
     {
       outputDirectory,
+      suiteNameTemplate: `${process.env.TURBOPACK ? 'turbo-':''}${process.env.EXPERIMENTAL_TURBOPACK ? 'experimentalTurbo-': ''}${process.env.NEXT_TEST_MODE}-{filename}-{title}`,
```

Comment:
this is really helpful in datadog, but it would be cool to use the full pathname (relative to ./test) instead of `filename` only, since it's often `index.test.ts`.

Reply:
yes, could use full path instead easily. unit test having =dev is probably not something amendable in this PR though, as CI is already configured in such was. Those env marker is needed since same test case run with different test mode, without specifying it test results will be merged as retry attempt instead of separate run.